crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -297,7 +390,34 @@ impl<F>
                     connector_metadata: None,
                     network_txn_id: None,
                     connector_response_reference_id: None,
-                    incremental_authorization_allowed: None,
+                    incremental_authorization_allowed: Some(
+                        status == enums::AttemptStatus::Authorized,
+                    ),
+                    charge_id: None,
+                })
+            }
+            DatatransResponse::ThreeDSResponse(response) => {
+                let redirection_link = match item.data.test_mode {
+                    Some(true) => format!("{}/v1/start", REDIRECTION_SBX_URL),
+                    Some(false) | None => format!("{}/v1/start", REDIRECTION_PROD_URL),
+                };
+                let redirection_data = Some(services::RedirectForm::Form {
+                    endpoint: format!("{}/{}", redirection_link, response.transaction_id),
+                    method: services::Method::Get,
+                    form_fields: HashMap::new(),
+                });
+                Ok(types::PaymentsResponseData::TransactionResponse {
+                    resource_id: types::ResponseId::ConnectorTransactionId(
+                        response.transaction_id.clone(),
+                    ),
+                    redirection_data,
+                    mandate_reference: None,
+                    connector_metadata: None,
+                    network_txn_id: None,
+                    connector_response_reference_id: None,
+                    incremental_authorization_allowed: Some(
+                        status == enums::AttemptStatus::Authorized,
+                    ),
```

Comment:
This will always be false hence no need to pass it here.

Reply:
updated